### ISSUE TITLE
expected filesize "value" got "value" Error

### ISSUE THREAD

**Issue by jonasteixeira** [nextcloud/nextcloud-snap#593]
I have an install of the nextcloud 13.02 snap in ubuntu 16.04,
I have had some issues when some users try to upload some files in most videos. I have received the following message in nextcloud.log

{"reqId":"cTVQGr6E2di4ej0I2jXD","level":4,"time":"2018-05-23T22:23:43+00:00","remoteAddr":"179.83.179.212","user":"fabio.pinel","app":"webdav","method":"PUT","url":"\/remote.php\/webdav\/Material%20Bruto\/S%c3%a9culo%20XXI\/D%c3%a9cada%202010\/2018\/201805\/20180520\/sub-20%20rouparia5257.mov","message":"Exception: {\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\BadRequest\",\"Message\":\"expected filesize 289138230 got 214695936\",\"Code\":0,\"Trace\":\"#0 \\\/snap\\\/nextcloud\\\/6916\\\/htdocs\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Directory.php(151): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #20)\\n#1 \\\/snap\\\/nextcloud\\\/6916\\\/htdocs\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1096): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('sub-20 rouparia...', Resource id #20)\\n#2 \\\/snap\\\/nextcloud\\\/6916\\\/htdocs\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(525): Sabre\\\\DAV\\\\Server->createFile('Material Bruto\\\/...', Resource id #20, NULL)\\n#3 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#4 \\\/snap\\\/nextcloud\\\/6916\\\/htdocs\\\/3rdparty\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#5 \\\/snap\\\/nextcloud\\\/6916\\\/htdocs\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#6 \\\/snap\\\/nextcloud\\\/6916\\\/htdocs\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#7 \\\/snap\\\/nextcloud\\\/6916\\\/htdocs\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/webdav.php(80): Sabre\\\\DAV\\\\Server->exec()\\n#8 \\\/snap\\\/nextcloud\\\/6916\\\/htdocs\\\/remote.php(164): require_once('\\\/snap\\\/nextcloud...')\\n#9 {main}\",\"File\":\"\\\/snap\\\/nextcloud\\\/6916\\\/htdocs\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php\",\"Line\":180}","userAgent":"Mozilla\/5.0 (iOS) Nextcloud-iOS\/2.20.8","version":"13.0.2.1"}

I have done a lot of research in github's own forums and elsewhere, but in most cases, it is guided changes in PHP files that are presented as read-only in the nextcloud installed by snap.
I changed the upload size and memory size by snap command of nextcloud, but to no avail.
Has anyone else gone through this problem in cases of nextcloud installed by snap?

**Comment by r4co0n** [comment]
As you say this happened more often, do you see the same received bytes reported after `expected filesize <what was wanted> got `? I noticed that the data apparently gets chopped at precisely 199.75 MiB, which seems at least noteworthy.

Could you investigate if the behaviour persists when you do a PUT issued directly from the same machine the snap is running on?

**Comment by jonasteixeira** [comment]
Thank you for answering my question.

Even in cases where the same user tries to send the same files, the value received in "got" is different on each attempt, I have not seen repetition since value in the log.

I will try to perform the PUT from within the server itself, I will try using the nextcloud address with "localhost" or the server's own local network ip.
As soon as I get a return I'll inform you here.
Thanks in advance

**Comment by QuaySo** [comment]
See [#1450](https://github.com/nextcloud/android/issues/1450) where this exact same issue is discussed. They seem to have this fixed, but probably a recent update broke things again...

**Comment by Thorbinjho** [comment]
I'm having the exact same issue with my setup. I'm using an up-to-date Android App from the F-Droid store and my servers Nextcloud installation ist 13.0.2 from the Archlinux repositories. However, I'm going to update it to 13.0.4 within the next week, but as the issue seems to be caused by the client app, I'm not expecting to see any changes.
The issue occures in my case for any file that is bigger than 1 MB. Smaller files can be uploaded without any problem. This behaviour can be observed for manual as well as for auto uploading from within the App. Using the webinterface on my desktop PC, I can even upload files with a size of hundrets of Megabytes without any issues.

**Comment by jonasteixeira** [comment]
Thanks a lot for the help of everyone, but apparently the problem is in the internet link, which is failing during the download, since in the local network the transfer happens without major problems.
We changed the default internet firewall link from the network where the nextcloud server is hosted and had no problems.

**Comment by Thorbinjho** [comment]
Thank you for your feedback. Can you explain your solution a bit more clearly, @jonasteixeira ? My server is located somewhere remotely but I'm having a very good and fast connection to it. Downloading (even big) files or uploading small files <1MB with my mobile works without any problems and uploading big files within the webinterface from my desktop PC (different network than my server) works as well. I'm just facing the issue while trying to upload files >1MB from my mobile to the nextcloud server. I do not understand what you mean by "changing the default internet firewall link". Can you explain that a bit more detailed? My server is directly connected to the internet from within a server farm.

**Comment by jonasteixeira** [comment]
From what I noticed in your report, you're facing a more specific problem than mine. In my case in any scenario the upload was halted in the middle. By the agent on desktops, smartphones or the web interface. It was always interrupted.
In my case through the local network all the tests worked correctly.
In the network where my nextcloud server is installed it has a firewall (draytech) with 2 internet links, and the traffic started with a dedicated link that has a fixed ip, but it was traveling through the default internet connection of the firewall that was another one link that is flashing.
After "canceling" the problematic link of the firewall, configuring the access only by internet that has fixed ip the problem has been solved.
But from what I understand, your case is different, and I believe my solution would not apply.

**Comment by Thorbinjho** [comment]
Ok, than I agree that we do not have the same issue... :-(

**Comment by r4co0n** [comment]
@Thorbinjho, I'm using an up-to-date Nextcloud Android app from F-Droid as well, but for me, the upload is working for files >1MB, there must be something else going on here.

**Comment by Thorbinjho** [comment]
@r4co0n See the referenced thread (https://github.com/nextcloud/android/issues/1450) in the android-section. I figured out it was caused by the firewall on my mobile although I whitelisted the nextcloud app. I have no idea why it was working for the upload of small files, for downloads of any size and for my DavDroid while it did not work at all for files bigger than 1 MB. 

The app behaves absolutely normal as long as NetGuard is disabled...
